demo_nodes_py/demo_nodes_py/logging/use_logger_service.py

```diff
@@ -0,0 +1,180 @@
+        self._sub = self.create_subscription(String, 'output', self.callback, 10)
+
+    def callback(self, msg):
+        self.get_logger().debug(msg.data + ' log with DEBUG logger level.')
+        self.get_logger().info(msg.data + ' log with INFO logger level.')
+        self.get_logger().warn(msg.data + ' log with WARN logger level.')
+        self.get_logger().error(msg.data + ' log with ERROR logger level.')
+
+
+class TestNode(Node):
+
+    def __init__(self, remote_node_name):
+        super().__init__('TestNode')
+        self.pub = self.create_publisher(String, 'output', 10)
+        self._logger_get_client = self.create_client(
+            GetLoggerLevels, remote_node_name + '/get_logger_levels')
+        self._logger_set_client = self.create_client(
+            SetLoggerLevels, remote_node_name + '/set_logger_levels')
+        self._remote_node_name = remote_node_name
+
+    def set_logger_level_on_remote_node(self, logger_level) -> bool:
+        if not self._logger_set_client.service_is_ready():
+            return False
+
+        request = SetLoggerLevels.Request()
+        set_logger_level = LoggerLevel()
+        set_logger_level.name = self._remote_node_name
+        set_logger_level.level = logger_level
+        request.levels.append(set_logger_level)
+        SetLoggerLevels.Response
+
+        future = self._logger_set_client.call_async(request)
+        rclpy.spin_until_future_complete(self, future)
+
+        ret_results = future.result()
+        if not ret_results:
+            return False
+
+        if not ret_results.results[0].successful:
+            self.get_logger().error('Failed to change logger level: '
+                                    + ret_results.results[0].reason)
+            return False
+
+        return True
+
+    def get_logger_level_on_remote_node(self):
+        if not self._logger_get_client.service_is_ready():
+            return [False, None]
+
+        request = GetLoggerLevels.Request()
+        request.names.append(self._remote_node_name)
+
+        future = self._logger_get_client.call_async(request)
+        rclpy.spin_until_future_complete(self, future)
+
+        ret_results = future.result()
+        if not ret_results:
+            return [False, None]
+
+        return [True, ret_results.levels[0].level]
+
+
+def get_logger_level_func(test_node):
+    ret, level = test_node.get_logger_level_on_remote_node()
+    if ret:
+        test_node.get_logger().info('Current logger level: ' + str(level))
+    else:
+        test_node.get_logger().error('Failed to get debug logger level via logger service !')
+
+
+def main(args=None):
+    rclpy.init(args=args)
+
+    logger_service_node = LoggerServiceNode()
+    test_node = TestNode('LoggerServiceNode')
+
+    executor = SingleThreadedExecutor()
+    executor.add_node(logger_service_node)
+
+    def thread_process(node):
+        executor.spin()
+
+    thread = threading.Thread(target=thread_process, args=(executor,))
```

Comment:
```suggestion
    thread = threading.Thread(target=thread_process)
```

Reply:
Oh, using `threading.Thread(target=executor.spin)` directly might be better.

Reply:
Yeah. It is better.